src/caching_client.rs

```diff
@@ -73,24 +72,22 @@ impl CachingClient {
   }
 
   #[inline]
-  fn debuginfo_path(&self, build_id: &[u8]) -> PathBuf {
-    let build_id = format_build_id(build_id);
-
-    self.cache_dir.join(build_id).join("debuginfo")
+  fn debuginfo_path(&self, build_id: &BuildId) -> PathBuf {
+    self.cache_dir.join(build_id.formatted()).join("debuginfo")
   }
 
   /// Fetch the debug info for the given build ID.
-  pub fn fetch_debug_info(&self, build_id: &[u8]) -> Result<Option<PathBuf>> {
+  pub fn fetch_debug_info(&self, build_id: &BuildId) -> Result<Option<PathBuf>> {
     let path = self.debuginfo_path(build_id);
     if path.try_exists()? {
       debug!("cache hit on `{}`", path.display());
-      return Ok(Some(path))
+      return Ok(Some(path));
```

Comment:
Seems like my local clippy added this, fwiw I am running

```
[javierhonduco@fedora debuginfod]$ cargo clippy --version
clippy 0.1.81 (eeb90cd 2024-09-04)
```

Reply:
I think it's `rustfmt`, really. A bunch of the [options](https://github.com/d-e-s-o/debuginfod/blob/main/.rustfmt.toml) I set are only honored by the nightly version. Any change you could install that or at least remove the spurious changes from the PR?

Reply:
You may also get away with just setting `RUSTC_BOOTSTRAP=1` for the `rustfmt` invocation. I don't know if it's honored by the program, but in `rustc` it's a hack to enable nightly code paths on stable.